tools/donate-cpu.py

```diff
@@ -142,10 +147,22 @@
         sys.exit(1)
     for ver in cppcheck_versions:
         if ver == 'head':
-            if not compile_cppcheck(cppcheck_path, jobs):
-                print('Failed to compile Cppcheck, retry later')
-                sys.exit(1)
-        elif not compile_version(work_path, jobs, ver):
+            ver = 'main'
+        current_cppcheck_dir = os.path.join(work_path, 'tree-'+ver)
+        try:
+            print('Fetching Cppcheck-{}..'.format(ver))
+            try_retry(get_cppcheck_version, fargs=(repo_path, ver, current_cppcheck_dir))
+        except KeyboardInterrupt as e:
+            # Passthrough for user abort
+            raise e
+        except:
+            print('Failed to update Cppcheck, retry later')
+            sys.exit(1)
+        print('Compiling Cppcheck-{}..'.format(ver))
+        if ver == 'main' and not compile_cppcheck(current_cppcheck_dir, jobs):
```

Comment:
I will move this compile_cppcheck() call within the if block to avoid duplicates compile_cppcheck+compile_version calls for main. 
Will wait for further comments before I update this PR with another commit.

```
        if ver == 'main':
            if not compile_cppcheck(current_cppcheck_dir, jobs):
                print('Failed to compile Cppcheck-{}, retry later'.format(ver))
                sys.exit(1)
        else:
            if not compile_version(current_cppcheck_dir, jobs):
                print('Failed to compile Cppcheck-{}, retry later'.format(ver))
                sys.exit(1)
```